From 87e10f2a3b5d9867c104ffb4ed023bff0a64f218 Mon Sep 17 00:00:00 2001
From: Brian Harring <ferringb@gmail.com>
Date: Thu, 17 Nov 2011 16:27:30 -0800
Subject: [PATCH] allow dosym some-file directory/ despite PMS

This conflicts with PMS (gentoo bug 379899), but neither the usage nor
spec is being corrected to rectify it.  Rather than break things for
users, we now allow it.
---
 pkgcore/ebuild/eapi-bash/helpers/common/dohard |    7 ++++---
 1 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/pkgcore/ebuild/eapi-bash/helpers/common/dohard b/pkgcore/ebuild/eapi-bash/helpers/common/dohard
index 13ebeda..d00946f 100755
--- a/pkgcore/ebuild/eapi-bash/helpers/common/dohard
+++ b/pkgcore/ebuild/eapi-bash/helpers/common/dohard
@@ -8,9 +8,10 @@ check_args 2
 ln_option=
 src_prefix=
 if [[ ${HELPER_NAME} == dosym ]]; then
-	# T while nonportable, is to ensure that if there is a directory
-	# conflict, it doesn't just create the sym w/in
-	ln_option=snT
+	ln_option=sn
+	if [[ ${2} == */ ]] || [[ -d $2 && ! -L $2 ]]; then
+		echo "QA: dosym target '$2' omits the filename target; see bug 379899" >&2
+	fi
 else
 	src_prefix="${D}"
 fi
-- 
1.7.8.rc1

